src/main/java/de/craften/plugins/rpgplus/components/entitymanager/RpgPlusEntity.java

```diff
@@ -43,22 +147,31 @@ public NPC getNpc() {
     }
 
     public String getName() {
-        return npc.getName();
+        return name;
     }
 
     public void setName(String name) {
-        npc.setName(name);
+    	this.name = name;
+    	if (type == EntityType.PLAYER) {
+    	   npc.setName(name);
+       } else {
+    	   if (nametag != null) {
+    		   nametag.setCustomName(name);
+    	   }
+       }
```

Comment:
This should also use the `NameTrait`, same for the setter and the getter/setter of `secondName`.